src/Units/KaijuWarsWeapons.java

```diff
@@ -0,0 +1,432 @@
+        negateCounterBonuses = true;
+    }
+    protected KaijuWarsWeapon(int vsLand, int vsAir)
+    {
+      this(vsLand, vsAir, 1, 1);
+    }
+    public KaijuWarsWeapon(KaijuWarsWeapon other)
+    {
+      this(other.vsLand, other.vsAir, other.rangeMin, other.rangeMax);
+      canFireAfterMoving = other.canFireAfterMoving;
+    }
+    @Override
+    public WeaponModel clone()
+    {
+      return new KaijuWarsWeapon(this);
+    }
+
+    @Override
+    public double getDamage(KaijuWarsUnitModel defender)
+    {
+      int attack = deriveAttack(this, defender);
+      if( defender.isKaiju )
+        return attack * 10;
+
+      int counterPower = deriveCounter(this, defender);
+
+      return KaijuWarsWeapons.getDamage(attack, counterPower);
+    }
+
+    @Override
+    public double getDamage(TerrainType target)
+    {
+      if( TerrainType.METEOR == target )
+        return KaijuWarsWeapons.getDamage(vsLand, TERRAIN_DURABILITY);
+      return 0;
+    }
+  }
+
+  // generic land
+  public static class ShootLand extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public ShootLand(int power)
+    {
+      super(power, 0);
+    }
+  }
+  public static class ShootAll extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public ShootAll(int power)
+    {
+      super(power, power);
+    }
+  }
+
+  // specific land
+  public static class Missile extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public Missile()
+    {
+      super(2, 3);
+    }
+  }
+  // Missiles are really bad, so these get to be "buffed Missile" because just giving them this buff would be busted
+  public static class Rockets extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+    private static final int MIN_RANGE = 1;
+    private static final int MAX_RANGE = 2*KAIJU_SCALE_FACTOR;
+
+    public Rockets()
+    {
+      super(2, 3, MIN_RANGE, MAX_RANGE);
+      canFireAfterMoving = true;
+    }
+  }
+  public static class AA extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public AA()
+    {
+      super(1, 2);
+    }
+  }
+  public static class Artillery extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+    private static final int MIN_RANGE = 1;
+    private static final int MAX_RANGE = 3*KAIJU_SCALE_FACTOR;
+
+    public Artillery()
+    {
+      super(1, 0, MIN_RANGE, MAX_RANGE);
+    }
+  }
+
+  // air
+  public static class Fighter extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public Fighter()
+    {
+      super(1, 1);
+      isAirWeapon = true;
+    }
+  }
+  public static class Bomber extends KaijuWarsWeapon
+  {
+    private static final long serialVersionUID = 1L;
+
+    public Bomber()
+    {
+      super(2, 0);
+      isAirWeapon = true;
+    }
+  }
+
+  // experimental weapons
+  // TODO: slow logic?
```

Comment:
Is this comment still relevant?

Reply:
Technically, yes. The answer is "probably not, because I'm probably not implementing the unit"
I will rewrite it to just be a note to future me instead of a TODO, I think.